web/src/lightbox.ts

```diff
@@ -205,11 +205,13 @@ export function clear_for_testing(): void {
 }
 
 export function canonical_url_of_media(media: HTMLElement): string {
-    let media_src = media.getAttribute("src");
-    if (!media_src || media_src.startsWith("/user_uploads/thumbnail/")) {
-        media_src = media.parentElement!.getAttribute("href")!;
+    const media_src = media.getAttribute("src");
+    let media_url = media_src ? new URL(media_src, window.location.href).pathname : null;
+
+    if (!media_url || media_url?.startsWith("/user_uploads/thumbnail/")) {
+        media_url = media.parentElement!.getAttribute("href")!;
     }
```

Comment:
`.getAttribute("href")` doesn’t necessarily return the absolute URL.

We can use the `media.src` property and `media.parentElement.href` property, which the browser has already converted to absolute.

Reply:
I don't think that it's important that the return value be absolute?

Reply:
Well I think what's important is that it agree with other equivalent URLs, and converting everything to absolute seems like an effective approach?